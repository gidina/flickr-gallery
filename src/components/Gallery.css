.gallery {
  background: #eee;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(250px, 4fr));
  grid-gap: 10px;
  align-items: start;
}
