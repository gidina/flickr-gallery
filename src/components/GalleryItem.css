.gallery_item {
    width: 100%;
    height: 100%;
    background: #fff;
    border: 1px solid #f3f0f0;
    box-shadow: 0 1px 2px #afafaf;
    border-radius: 5px;
  }
  
  .gallery_item_preview {
    margin: 0.5em;
    transition: all 0.5s ease;
    position: relative;
    display: inline-block;
    overflow: hidden;
  }
  
  .gallery_item_preview svg.gallery_top {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 50px;
    transition: all 0.5s ease;
  }
  
  .gallery_item_preview button {
    position: absolute;
    top: 0.5em;
    left: -30em;
    background-color: #fff;
    color: #f57171;
    padding: 0.56em 1em;
    border-radius: 5px 0 0 0;
    transition: all 1s ease;
  }
  
  .gallery_item_preview img {
    width: 100%;
    height: 200px;
    border-radius: 5px;
  }
  
  .gallery_item_preview h3 {
    font-family: "Leckerli One", cursive;
  }
  
  .gallery_item_preview p {
    word-wrap: break-word;
    overflow: hidden;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
  }
  
  .gallery_item:hover {
    cursor: pointer;
    color: #777;
    transition: all 0.5s ease;
  }
  
  .gallery_item:hover svg.gallery_top {
    top: -10em;
    transition: all 0.5s ease;
  }
  
  .gallery_item:hover button {
    cursor: pointer;
    left: 0.5em;
    transition: all 1s ease;
  }