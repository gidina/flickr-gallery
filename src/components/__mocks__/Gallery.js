import React from "react";
export default class Gallery extends React.Component {
    render() {
        return <gallery-mock {...this.props}>{this.props.children}</gallery-mock>;
    }
}