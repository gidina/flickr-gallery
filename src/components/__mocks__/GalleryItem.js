import React from "react";
export default class GalleryItem extends React.Component {
    render() {
        return <gallery-item-mock {...this.props}>{this.props.children}</gallery-item-mock>;
    }
}
