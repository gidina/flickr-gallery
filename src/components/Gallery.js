import React, { Component, Fragment } from "react";
import { redColor, blueColor } from "../config";
import Pagination from "./Pagination";
import "./Gallery.css";

const breakpointDesktop = 1047;

class Gallery extends Component {
  state = {
    currentPhotos: this.props.photos,
    isDesktop: window.innerWidth > breakpointDesktop
  };
  componentDidMount = () => {
    window.addEventListener("resize", () => {
      const isDesktop = window.innerWidth > breakpointDesktop;
      const currentPhotos = isDesktop
        ? [...this.state.currentPhotos]
        : this.props.photos;
      this.setState({
        ...this.state,
        currentPhotos,
        isDesktop
      });
    });
  };
  onPageChange = (startIndex, endIndex) => {
    const { photos } = this.props;
    const currentPhotos = photos.slice(startIndex, endIndex + 1);
    this.setState({ ...this.state, currentPhotos });
  };
  render = () => {
    const { photos, onClickPhoto } = this.props;
    const { currentPhotos, isDesktop } = this.state;
    return (
      <Fragment>
        <div className="gallery">
          {currentPhotos.map(foto => (
            <div
              key={foto.id}
              className="gallery_item"
              onClick={() => onClickPhoto(foto)}
            >
              <span
                className="gallery_item_preview"
                onClick={() =>
                  this.setState({
                    ...this.state,
                    isModalOpen: true,
                    fotoSeleccionada: foto
                  })
                }
              >
                <button>Click to view</button>
                <svg
                  fill={foto.media === "video" ? blueColor : redColor}
                  className="gallery_top"
                  viewBox="0 3 60 20"
                >
                  <path d="M 0.65359477,1.3817905 C 60.201925,8.44316 121.92863,11.583451 175.81699,28.832771 l 0.6536,-28.10457531 z" />
                </svg>
                <img
                  src={`https://farm${foto.farm}.staticflickr.com/${
                    foto.server
                  }/${foto.id}_${foto.secret}_b.jpg`}
                  alt=""
                />
                <span>
                  <h3>{foto.title}</h3>
                  <p>{foto.description._content}</p>
                </span>
              </span>
            </div>
          ))}
        </div>
        {isDesktop ? (
          <Pagination
            numItems={photos.length}
            onPageChange={this.onPageChange}
          />
        ) : null}
      </Fragment>
    );
  };
}

// Gallery.protoTypes = {
//     photos: React.PropTypes.array.isRequired
// };

export default Gallery;
