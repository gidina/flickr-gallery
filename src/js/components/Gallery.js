import React from 'react';
import { redColor, blueColor } from "../../config";
import "./Gallery.css";

const midaFotos = "b";

const Gallery = ({photos, onClickPhoto}) => {
    return (
        <div className="gallery">

        
            {
                photos.map(foto => 
                    <div key={foto.id} className="gallery_item" onClick={() => onClickPhoto(foto)}>
                <span className="gallery_item_preview" onClick={() => this.setState({ ...this.state, isModalOpen: true, fotoSeleccionada: foto })}>
                    <button>Click to view</button>
                    <svg fill={foto.media === "video" ? blueColor : redColor} className="gallery_top" viewBox="0 3 60 20">
                        <path d="M 0.65359477,1.3817905 C 60.201925,8.44316 121.92863,11.583451 175.81699,28.832771 l 0.6536,-28.10457531 z"></path>
                    </svg>
                    <img src={`https://farm${foto.farm}.staticflickr.com/${foto.server}/${foto.id}_${foto.secret}_${midaFotos}.jpg`} alt="" />
                    <span>
                        <h3>{foto.title}</h3>
                        <p>{foto.description._content}</p>
                    </span>
                </span>
                </div>)
            }
                </div>

    );
}

// Gallery.protoTypes = {
//     photos: React.PropTypes.array.isRequired
// };

export default Gallery;