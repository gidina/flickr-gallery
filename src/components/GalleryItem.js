import React from "react";
import PropTypes from 'prop-types';
import { primaryColor, secondaryColor } from "../config";
import "./GalleryItem.css";

const GalleryItem = ({ photo, onClick }) => 
    <div className="gallery_item" onClick={onClick}>
        <span className="gallery_item_preview">
            <button>Click to view</button>
            <svg
                fill={photo.media === "video" ? secondaryColor : primaryColor}
                className="gallery_top"
                viewBox="0 3 60 20"
            >
                <path d="M 0.65359477,1.3817905 C 60.201925,8.44316 121.92863,11.583451 175.81699,28.832771 l 0.6536,-28.10457531 z" />
            </svg>
            <img
                src={`https://farm${photo.farm}.staticflickr.com/${photo.server}/${photo.id}_${photo.secret}_b.jpg`}
                alt=""
            />
            <span>
                <h3>{photo.title}</h3>
                <p>{photo.description._content}</p>
            </span>
        </span>
    </div>;

GalleryItem.propTypes = {
    photo: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired
};

export default GalleryItem;
