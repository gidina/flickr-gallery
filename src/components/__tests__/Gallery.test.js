import React from "react";
import renderer from "react-test-renderer";
import Gallery from "../Gallery";
import ReactTestUtils from "react-dom/test-utils";
import GalleryItem from "../GalleryItem";

jest.mock("../GalleryItem");
jest.mock("../Pagination");

window.resizeTo = (width, height) => {
    global.window.innerWidth = width || global.window.innerWidth;
    global.window.innerHeight = height || global.window.innerHeight;
    global.window.dispatchEvent(new Event("resize"));
};

const photosTest = [
    {
        id: 1,
        title: "Photo 1",
        description: { _content: "Description Photo 1" },
        media: "video",
        farm: 1,
        server: "server1",
        secret: 123456
    },
    {
        id: 2,
        title: "Photo 2",
        description: { _content: "Description Photo 2" },
        media: "video",
        farm: 1,
        server: "server1",
        secret: 456789
    },
    {
        id: 3,
        title: "Photo 3",
        description: { _content: "Description Photo 3" },
        media: "video",
        farm: 3,
        server: "server3",
        secret: 323456
    },
    {
        id: 4,
        title: "Photo 4",
        description: { _content: "Description Photo 4" },
        media: "video",
        farm: 1,
        server: "server1",
        secret: 456789
    },
    {
        id: 5,
        title: "Photo 5",
        description: { _content: "Description Photo 5" },
        media: "video",
        farm: 5,
        server: "server5",
        secret: 123456
    },
    {
        id: 6,
        title: "Photo 6",
        description: { _content: "Description Photo 6" },
        media: "video",
        farm: 1,
        server: "server1",
        secret: 456789
    },
    {
        id: 7,
        title: "Photo 7",
        description: { _content: "Description Photo 7" },
        media: "video",
        farm: 7,
        server: "server7",
        secret: 363456
    },
    {
        id: 8,
        title: "Photo 8",
        description: { _content: "Description Photo 8" },
        media: "video",
        farm: 1,
        server: "server1",
        secret: 856789
    },
    {
        id: 9,
        title: "Photo 9",
        description: { _content: "Description Photo 9" },
        media: "video",
        farm: 1,
        server: "server1",
        secret: 856789
    }
];

const breakpointDesktop = 1047;

test("Renders as many 'GalleryItem' as photos recived", () => {
    const tree = ReactTestUtils.renderIntoDocument(<Gallery photos={photosTest} onClickPhoto={() => {}} />);
    const galleryItems = ReactTestUtils.scryRenderedComponentsWithType(tree, GalleryItem);

    expect(galleryItems.length).toEqual(photosTest.length);
});

test(`Desktop resolution (>${breakpointDesktop}): 'Pagination' is rendered`, () => {
    window.resizeTo(breakpointDesktop+1, window.innerHeight);

    const tree = renderer
    .create(<Gallery photos={photosTest} onClickPhoto={() => {}} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

test(`Mobile or Tablet resolution (<=${breakpointDesktop}): 'Pagination' isn't rendered`, () => {
    window.resizeTo(breakpointDesktop, window.innerHeight);

    const tree = renderer
    .create(<Gallery photos={photosTest} onClickPhoto={() => {}} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

test("onClick GalleryItem, argument: photo", () => {
    const onClickPhotoMock = jest.fn();
    const indexPhotoSelected = 0;

    const tree = ReactTestUtils.renderIntoDocument(<Gallery photos={photosTest} onClickPhoto={onClickPhotoMock} />);
    const galleryItems = ReactTestUtils.scryRenderedComponentsWithType(tree, GalleryItem);

    expect(onClickPhotoMock).toHaveBeenCalledTimes(0);
    galleryItems[indexPhotoSelected].props.onClick();

    expect(onClickPhotoMock).toHaveBeenCalledTimes(1);
    expect(onClickPhotoMock).toHaveBeenCalledWith(photosTest[indexPhotoSelected]);
});

// test("Renders correctly with photo properties. If photo.media is different to 'video', svg must be red (#f55)", () => {
//     const photoTest = {
//         id: 1,
//         title: "Photo 1",
//         description: { _content: "Description Photo 1" },
//         media: "photo",
//         farm: 1,
//         server: "server1",
//         secret: 123456
//     };
//     const tree = renderer
//         .create(<Gallery photo={photoTest} onClick={() => { }} />)
//         .toJSON();
//     expect(tree).toMatchSnapshot();
// });

// test("onClick function is triggered when div with 'gallery_item' class is clicked", () => {
//     const onClickMock = jest.fn();
//     const photoTest = {
//         id: 1,
//         title: "Photo 1",
//         description: { _content: "Description Photo 1" },
//         media: "video",
//         farm: 1,
//         server: "server1",
//         secret: 123456
//     };

//     class Wrapper extends React.Component {
//         render() {
//             return this.props.children
//         }
//     }

//     const tree = ReactTestUtils.renderIntoDocument(<Wrapper><Gallery photo={photoTest} onClick={onClickMock} /></Wrapper>);

//     expect(tree.props.children.props.onClick).toEqual(onClickMock);

//     // // trigger the onClick event for the Gallery
//     // tree.props.children.props.onClick();
//     // expect(onClickMock).toHaveBeenCalledWith();
// });

// // describe('the simplest way to test a react component', () => {
// //     it ('mount should render a simple component', () => {
// //         let component = mount(Bar, { title: 'fooBar' });
// //         let node = findDOMNode(component);
// //         expect(node.textContent).toEqual('fooBar')
// //     });
// // });

// // import React from 'react';
// // import { shallow } from 'enzyme';
// // import Button from './Button';

// // describe('Test Button component', () => {
// //   it('Test click event', () => {
// //     const mockCallBack = jest.fn();

// //     const button = shallow((<Button onClick={mockCallBack}>Ok!</Button>));
// //     button.find('button').simulate('click');
// //     expect(mockCallBack.mock.calls.length).toEqual(1);
// //   });
// // });