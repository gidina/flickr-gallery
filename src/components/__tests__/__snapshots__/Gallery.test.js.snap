// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop resolution (>1047): 'Pagination' is rendered 1`] = `
Array [
  <div
    className="gallery"
  >
    <gallery-item-mock
      onClick={[Function]}
      photo={
        Object {
          "description": Object {
            "_content": "Description Photo 1",
          },
          "farm": 1,
          "id": 1,
          "media": "video",
          "secret": 123456,
          "server": "server1",
          "title": "Photo 1",
        }
      }
    />
    <gallery-item-mock
      onClick={[Function]}
      photo={
        Object {
          "description": Object {
            "_content": "Description Photo 2",
          },
          "farm": 1,
          "id": 2,
          "media": "video",
          "secret": 456789,
          "server": "server1",
          "title": "Photo 2",
        }
      }
    />
    <gallery-item-mock
      onClick={[Function]}
      photo={
        Object {
          "description": Object {
            "_content": "Description Photo 3",
          },
          "farm": 3,
          "id": 3,
          "media": "video",
          "secret": 323456,
          "server": "server3",
          "title": "Photo 3",
        }
      }
    />
    <gallery-item-mock
      onClick={[Function]}
      photo={
        Object {
          "description": Object {
            "_content": "Description Photo 4",
          },
          "farm": 1,
          "id": 4,
          "media": "video",
          "secret": 456789,
          "server": "server1",
          "title": "Photo 4",
        }
      }
    />
    <gallery-item-mock
      onClick={[Function]}
      photo={
        Object {
          "description": Object {
            "_content": "Description Photo 5",
          },
          "farm": 5,
          "id": 5,
          "media": "video",
          "secret": 123456,
          "server": "server5",
          "title": "Photo 5",
        }
      }
    />
    <gallery-item-mock
      onClick={[Function]}
      photo={
        Object {
          "description": Object {
            "_content": "Description Photo 6",
          },
          "farm": 1,
          "id": 6,
          "media": "video",
          "secret": 456789,
          "server": "server1",
          "title": "Photo 6",
        }
      }
    />
    <gallery-item-mock
      onClick={[Function]}
      photo={
        Object {
          "description": Object {
            "_content": "Description Photo 7",
          },
          "farm": 7,
          "id": 7,
          "media": "video",
          "secret": 363456,
          "server": "server7",
          "title": "Photo 7",
        }
      }
    />
    <gallery-item-mock
      onClick={[Function]}
      photo={
        Object {
          "description": Object {
            "_content": "Description Photo 8",
          },
          "farm": 1,
          "id": 8,
          "media": "video",
          "secret": 856789,
          "server": "server1",
          "title": "Photo 8",
        }
      }
    />
    <gallery-item-mock
      onClick={[Function]}
      photo={
        Object {
          "description": Object {
            "_content": "Description Photo 9",
          },
          "farm": 1,
          "id": 9,
          "media": "video",
          "secret": 856789,
          "server": "server1",
          "title": "Photo 9",
        }
      }
    />
  </div>,
  <pagination-mock
    numItems={9}
    onPageChange={[Function]}
  />,
]
`;

exports[`Mobile or Tablet resolution (<=1047): 'Pagination' isn't rendered 1`] = `
<div
  className="gallery"
>
  <gallery-item-mock
    onClick={[Function]}
    photo={
      Object {
        "description": Object {
          "_content": "Description Photo 1",
        },
        "farm": 1,
        "id": 1,
        "media": "video",
        "secret": 123456,
        "server": "server1",
        "title": "Photo 1",
      }
    }
  />
  <gallery-item-mock
    onClick={[Function]}
    photo={
      Object {
        "description": Object {
          "_content": "Description Photo 2",
        },
        "farm": 1,
        "id": 2,
        "media": "video",
        "secret": 456789,
        "server": "server1",
        "title": "Photo 2",
      }
    }
  />
  <gallery-item-mock
    onClick={[Function]}
    photo={
      Object {
        "description": Object {
          "_content": "Description Photo 3",
        },
        "farm": 3,
        "id": 3,
        "media": "video",
        "secret": 323456,
        "server": "server3",
        "title": "Photo 3",
      }
    }
  />
  <gallery-item-mock
    onClick={[Function]}
    photo={
      Object {
        "description": Object {
          "_content": "Description Photo 4",
        },
        "farm": 1,
        "id": 4,
        "media": "video",
        "secret": 456789,
        "server": "server1",
        "title": "Photo 4",
      }
    }
  />
  <gallery-item-mock
    onClick={[Function]}
    photo={
      Object {
        "description": Object {
          "_content": "Description Photo 5",
        },
        "farm": 5,
        "id": 5,
        "media": "video",
        "secret": 123456,
        "server": "server5",
        "title": "Photo 5",
      }
    }
  />
  <gallery-item-mock
    onClick={[Function]}
    photo={
      Object {
        "description": Object {
          "_content": "Description Photo 6",
        },
        "farm": 1,
        "id": 6,
        "media": "video",
        "secret": 456789,
        "server": "server1",
        "title": "Photo 6",
      }
    }
  />
  <gallery-item-mock
    onClick={[Function]}
    photo={
      Object {
        "description": Object {
          "_content": "Description Photo 7",
        },
        "farm": 7,
        "id": 7,
        "media": "video",
        "secret": 363456,
        "server": "server7",
        "title": "Photo 7",
      }
    }
  />
  <gallery-item-mock
    onClick={[Function]}
    photo={
      Object {
        "description": Object {
          "_content": "Description Photo 8",
        },
        "farm": 1,
        "id": 8,
        "media": "video",
        "secret": 856789,
        "server": "server1",
        "title": "Photo 8",
      }
    }
  />
  <gallery-item-mock
    onClick={[Function]}
    photo={
      Object {
        "description": Object {
          "_content": "Description Photo 9",
        },
        "farm": 1,
        "id": 9,
        "media": "video",
        "secret": 856789,
        "server": "server1",
        "title": "Photo 9",
      }
    }
  />
</div>
`;
